ginga/util/iohelper.py

```diff
@@ -18,29 +18,48 @@ def get_fileinfo(filespec, cache_dir='/tmp', download=False):
     """
     Parse a file specification and return information about it.
     """
-    idx = None
-    name_ext = ''
+    # Loads first science extension by default.
+    # This prevents [None] to be loaded instead.
+    idx = ('SCI', 1)
+    name_ext = '[SCI,1]'
 
     # User specified an HDU using bracket notation at end of path?
-    match = re.match(r'^(.+)\[(.+)\]$', filespec)
+
+    # This matches
+    #   image.fits[("NAME", 1)]
+    #   image.fits[('NAME', 1)]
+    #   image.fits['NAME', 1]
+    #   image.fits["NAME", 1]
+    #   image.fits[NAME, 1]
+    #   image.fits[(NAME, 1)]
+    match = re.match(r'^(.+)\[\(?[\'\"]?([A-Za-z]+)[\'\"]?\,([0-9])\)?\]$',
+                     filespec)
```

Comment:
@pllim, I just want to make sure that we cover all the cases for naming because I'm sure I'll get a bug report from someone that has a strangely named HDU.  FITS is terrible that way.